loki/transform/transform_hoist_variables.py

```diff
@@ -131,7 +132,16 @@ def transform_subroutine(self, routine, **kwargs):
 
         role = kwargs.get('role', None)
         item = kwargs.get('item', None)
-        successors = kwargs.get('successors', None)
+        _successors = kwargs.get('successors', None)
+        successors = []
+        for successor in _successors:
+            append = True
+            for _disable in self.disable:
+                if _disable.upper() in successor.name.upper():
```

Comment:
Yes, for CLOUDSC config:

```py
disable = ['timer%start', 'timer%end', 'timer%thread_start', ...]
```

as within the CLOUDSC driver

```fortran
...
CALL TIMER%START(NUMOMP)
...    
TID = GET_THREAD_NUM()
CALL TIMER%THREAD_START(TID)
...
```

however

```py
for successor in _successors:
      append = True
      for _disable in self.disable:
          if _disable.upper() in successor.name.upper():
              print(f"{_disable.upper()} vs. {successor.name.upper()}")  
```

gives: 

```
TIMER%START vs. TIMER_MOD#PERFORMANCE_TIMER%START
TIMER%THREAD_START vs. TIMER_MOD#PERFORMANCE_TIMER%THREAD_START
TIMER%THREAD_LOG vs. TIMER_MOD#PERFORMANCE_TIMER%THREAD_LOG
TIMER%THREAD_END vs. TIMER_MOD#PERFORMANCE_TIMER%THREAD_END
TIMER%END vs. TIMER_MOD#PERFORMANCE_TIMER%END
TIMER%PRINT_PERFORMANCE vs. TIMER_MOD#PERFORMANCE_TIMER%PRINT_PERFORMANCE
```

therefore, I would need to specify in the config something like 

```py
disable = ["TIMER_MOD#PERFORMANCE_TIMER%START", ...]
```

in order to use `==` instead of `in`?

Reply:
Ah, I see. I think you need to use `successor.local_name`. This gives you the item name without the scope.

Reply:
Ok, still 

with 

```fortran
TYPE(PERFORMANCE_TIMER) :: TIMER
...
CALL TIMER%THREAD_START(TID)
...
```

```
# print(f"{_disable.upper()} vs. {successor.local_name.upper()}")
TIMER%START vs. PERFORMANCE_TIMER%START
TIMER%THREAD_START vs. PERFORMANCE_TIMER%THREAD_START
TIMER%THREAD_LOG vs. PERFORMANCE_TIMER%THREAD_LOG
TIMER%THREAD_LOG vs. PERFORMANCE_TIMER%THREAD_LOG
TIMER%THREAD_END vs. PERFORMANCE_TIMER%THREAD_END
TIMER%END vs. PERFORMANCE_TIMER%END
TIMER%PRINT_PERFORMANCE vs. PERFORMANCE_TIMER%PRINT_PERFORMANCE
```

I guess it depends on the requirements on `disable`. Accordingly, I would leave the decision to you.

Reply:
Yes

Reply:
Perfectly fine for me. In the end, it's just a coincidence that it works with `disable = ["TIMER%START", ...]`.

Therefore, I would go with: 

```py
for successor in _successors:
    if not any(_disable == successor.local_name.upper() for _disable in self.disable):
        successors.append(successor)
```

whereas `self.disable = [_.upper() for _ in disable]`